python/log_closest_obstacle.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+import rospy
+from geopy.distance import distance
+from local_pathfinding.msg import AISMsg, GPS
+import os
+from datetime import datetime
+from datetime import date
+import pickle
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+
+UPDATE_TIME_SECONDS = 1.0
+
+# Output paths
+dateStr = date.today().strftime("%b-%d-%Y")
+timeStr = datetime.now().strftime('%H-%M-%S')
+ABS_PATH_TO_THIS_FILE_DIR = os.path.dirname(os.path.abspath(__file__))
+ABS_PATH_TO_OUTPUT_DIR = os.path.join(ABS_PATH_TO_THIS_FILE_DIR, "../output/{}_{}".format(dateStr, timeStr))
+ABS_PATH_TO_OUTPUT_MIN_DISTANCE_FILE = os.path.join(ABS_PATH_TO_OUTPUT_DIR, "minDistance.pkl")
+ABS_PATH_TO_OUTPUT_MIN_DISTANCE_PLOT_FILE = os.path.join(ABS_PATH_TO_OUTPUT_DIR, "min_distance_plot.png")
+
+
+class LogClosestObstacle:
+    def __init__(self):
+        rospy.init_node('log_closest_obstacle', anonymous=True)
+        rospy.Subscriber('/AIS', AISMsg, self.AIS_callback)
+        rospy.Subscriber('/GPS', GPS, self.GPS_callback)
+        self.ships = rospy.wait_for_message('/AIS', AISMsg).ships
+        self.lat = rospy.wait_for_message('/GPS', GPS).lat
+        self.lon = rospy.wait_for_message('/GPS', GPS).lon
+
+        self.closestDistances = list()
+
+    def get_closest_distances(self):
+        return self.closestDistances
+
+    def AIS_callback(self, data):
+        self.ships = data.ships
+
+    def GPS_callback(self, data):
+        self.lat = data.lat
+        self.lon = data.lon
+
+    def find_closest_ship(self):
+        closest_ship_dist = None if len(self.ships) == 0 else min([distance((ship.lat, ship.lon),
+                                                                            (self.lat, self.lon)).km
+                                                                  for ship in self.ships])
```

Comment:
This might be hard to understand, I just wanted to try out the list comprehension thing. If you want, I can change it back to a normal for loop.

Reply:
Or maybe I could make a distance function?

Reply:
Added distance function